### DISCUSSION TITLE
How to push personal access token of GitHub to my repo and keep that token do not expire

### DISCUSSION THREAD

**Discussion by gaomingzhao666** [community/community#126371] [Actions] [answered]
### Select Topic Area

Question

### Body

**My personal access token of GitHub is just permitted read-only to my repos, and I am making a dev portfolio for personal use.**
 
**The problem is the token will automatically and suddenly expire when I push the token to my repo, I want to know how to solve it because the portfolio relies on this token to display the projects I made, kindly help!!**

**Comment by gaomingzhao666** [comment, marked as answer]
**I tried to split the string of this token and then combine that string, then this problem be solved.**

**Comment by Pelaingi** [comment]
.github/workflows/ci.yml

**Comment by HUAWE33** [comment]
I tried to split the string of this token and then combine that string, then this problem be solved.

**Comment by imDarshanGK** [comment]
Store your GitHub token in environment variables and avoid pushing it to your repository to prevent automatic expiration and ensure security.

**Reply by gaomingzhao666** [reply]
> Store your GitHub token in environment variables and avoid pushing it to your repository to prevent automatic expiration and ensure security.

Thank you for your reply! But I made this website for the reason that shows my projects, I am not sure that the env variables can be build at bundle time, so I still using the solution above at this time.
